akka-http/src/main/scala/akka/http/scaladsl/server/directives/FutureDirectives.scala

```diff
@@ -32,6 +38,25 @@ trait FutureDirectives {
     }
 
   /**
+    * "Unwraps" a `Future[T]` and runs the inner route after future
+    * completion with the future's value as an extraction of type `T` if
+    * the supplied `CircuitBreaker` is open.
+    * If the supplied `CircuitBreaker` is open the request is rejected
+    * with a `CircuitBreakerOpenRejection`.
+    *
+   * @group future
+   */
+  def onCompleteWithBreaker[T](breaker: CircuitBreaker)(future: ⇒ Future[T]): Directive1[Try[T]] =
+    onComplete(breaker.withCircuitBreaker(future)).flatMap {
+      case Failure(ex: CircuitBreakerOpenException) ⇒
+        extractRequestContext.flatMap { ctx ⇒
+          ctx.request.entity.dataBytes.runWith(Sink.cancelled)(ctx.materializer)
+          reject(CircuitBreakerOpenRejection(ex))
+        }
+      case x ⇒ provide(x)
+    }
```

Comment:
Integrating well with circuit breakers is something I'd be willing to accept into core – it's a bit funny how people don't even realise we do have them, so by having it here we'd promote the use of breakers (which is good).

Do you strongly oppose or just wonder? :)